src/Entity/Notifications/Notification.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Entity\Notifications;
+
+use App\Entity\User;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\Mapping\InheritanceType;
+use Doctrine\ORM\Mapping\DiscriminatorMap;
+use Doctrine\ORM\Mapping\DiscriminatorColumn;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity
+ * @InheritanceType("SINGLE_TABLE")
+ * @DiscriminatorColumn(name="type", type="string")
+ * @DiscriminatorMap({
+ *      "submitAdded" = "SubmitAddedNotification",
+ *      "submitStatusChanged" = "SubmitStatusChangedNotification",
+ *      "participationStatusChangedNotification" = "ParticipationStatusChangedNotification",
+ *      "newFeaturedConference" = "NewFeaturedConferenceNotification"
+ * })
+ */
+abstract class Notification
+{
+    // These const must match the names of the templates in templates/user/notifications
+    public const TRIGGER_SUBMIT_ADDED = 'submit_added';
+    public const TRIGGER_SUBMIT_STATUS_CHANGED = 'submit_status_changed';
+    public const TRIGGER_PARTICIPATION_STATUS_CHANGED = 'participation_status_changed';
+    public const TRIGGER_NEW_FEATURED_CONFERENCE = 'new_featured_conference';
+
+    public const TRIGGERS = [
+        self::TRIGGER_SUBMIT_ADDED,
+        self::TRIGGER_SUBMIT_STATUS_CHANGED,
+        self::TRIGGER_PARTICIPATION_STATUS_CHANGED,
+        self::TRIGGER_NEW_FEATURED_CONFERENCE,
+    ];
+
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     * @ORM\Column(type="integer")
+     */
+    protected int $id;
+
+    /**
+     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="notifications")
+     * @ORM\JoinColumn(nullable=false)
+     */
+    protected User $targetUser;
```

Comment:
why `targetUser` ? `user` is enought, isn't?

Reply:
Since we also have an `emitter` on some child entities who is a user as well, it's more clear (in my opinion) to have a bit more precisions about who does what